### PR TITLE
22 connection

### PR DESCRIPTION
**Pull request by Herelius**

### CODE REVIEW

**Review by hkamalo — COMMENTED** [review]
Ok for me - wait for the other #22 issue to finish, merge them both before merging to dev

File: app.js
Comment (edouardmisset) [inline app.js]: Why renaming it expressSession and not keep session ?

File: app.js
Comment (edouardmisset) [inline app.js]: Follow-up from above: is it the right name ? 
Isn't it a connection instead... (postgreConnection maybe ?)

File: db.js
Comment (edouardmisset) [inline db.js:26]: Why do we export Prisma AND db ?
Shouldn't we simply export a connexion to the db like Pierre ?

BTW : Pierre's db.js file is 3 lines :
```javascript
const { PrismaClient } = require('@prisma/client');

const prisma = new PrismaClient();

module.exports = prisma;
```

File: models/ExperimentModel.js
Comment (edouardmisset) [inline models/ExperimentModel.js:1]: Delete it instead of commenting out 😉 
```suggestion

```

File: models/LocationModel.js
Comment (edouardmisset) [inline models/LocationModel.js:1]: Delete it instead of commenting out 😉 
```suggestion

```

File: models/UserModel.js
Comment (edouardmisset) [inline models/UserModel.js]: Delete it instead of commenting out 😉 
```suggestion

```

File: models/UserModel.js
Comment (edouardmisset) [inline models/UserModel.js:15]:
```suggestion
// Function to validate the unicity of a username
```

File: models/UserModel.js
Comment (edouardmisset) [inline models/UserModel.js:17]: [Prisma's doc](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique)
```suggestion
  !!(await prisma.user.findUnique({ where: { username } }));
```

File: models/UserModel.js
Comment (edouardmisset) [inline models/UserModel.js]: [Prisma's doc](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique)

```suggestion
await  prisma.user.findUnique({ where: { username } });
```

File: models/UserModel.js
Comment (edouardmisset) [inline models/UserModel.js:31]: One seems really short for a minimum password length 😉 

```suggestion
    password: Joi.string().min(4).max(100).required(),
```

File: models/UserModel.js
Comment (edouardmisset) [inline models/UserModel.js:38]:
```suggestion
await  return prisma.user.create({
```

File: routes/currentUser.js
Comment (edouardmisset) [inline routes/currentUser.js]: Why the console log ? dead code ?

```suggestion

```

File: routes/currentUser.js
Comment (edouardmisset) [inline routes/currentUser.js]: Probably need to wrap it in a getSafeAttributes() function like Pierre did.
Also, need to create this function in the model 😉 

```suggestion
    res.json(User.getSafeAttributes(req.currentUser));
```

File: routes/index.js
Comment (edouardmisset) [inline routes/index.js:14]: Why all this dead code ??

**Review by edouardmisset — CHANGES_REQUESTED** [review]
@Herelius : A few comments to look at

**Pull request closed without merging**
